c7n/resources/rds.py

```diff
@@ -2182,3 +2182,34 @@ def process(self, resources, event=None):
                     break
 
         return results
+
+
+@filters.register('has-pending-maintenance')
+class HasPendingMaintenance(Filter):
+    """ Scan DB instances for those with pending maintenance
+    :example:
+    .. code-block:: yaml
+            policies:
+              - name: rds-pending-maintenance
+                resource: rds
+                filters:
+                  - has-pending-maintenance
+    """
+
+    schema = type_schema('has-pending-maintenance')
+    permissions = ('rds:DescribePendingMaintenanceActions',)
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('rds')
+
+        results = []
+        response = client.describe_pending_maintenance_actions()
+        rds_pending_maintenance_list = [x['ResourceIdentifier']
+                                        for x in response.get('PendingMaintenanceActions')]
```

Comment:
This call can return up to 100 instances, but it's possible to have more than that so we'll probably want to cover that case. One possible approach:
 
```suggestion
        pending_maintenance = set()
        paginator = client.get_paginator('describe_pending_maintenance_actions')
        for page in paginator.paginate():
            pending_maintenance.update(
                {action['ResourceIdentifier'] for action in page['PendingMaintenanceActions']}
            )
```

This also tracks IDs in a set which makes membership checks a bit smoother later. Another option might be to call `describe_pending_maintenance_actions()` with a `db-instance-id` filter as mentioned [here](https://github.com/cloud-custodian/cloud-custodian/pull/1904/files#r157669992). Though there we'd need to break the resource set up into chunks of 100 because you can only filter for 100 unique DB instance IDs per call.

Either way seems reasonable to me - but I _think_ paginating the pending maintenance check is a bit more obvious from the reader perspective :thinking: .